executor/bitmap.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+#include <fcntl.h>
+#include <stdio.h>
+
+#if GOOS_linux || GOOS_freebsd || GOOS_netbsd || GOOS_openbsd || GOOS_akaros
+#include <sys/mman.h>
+#endif
+
+static byte* bitmap;
+static uint32 pcstart;
+static uint32 pcsize;
+static uint32 pcend;
+
+inline void init_bitmap()
+{
+#if SYZ_EXECUTOR_USES_SHMEM
+	int f = open("/syz-cover-bitmap", O_RDONLY);
+	if (f < 0) {
+		debug("bitmap is no found, coverage filter disabled\n");
+		return;
+	}
+	// If bitmap is existed, but invalid, executor should exit with error.
+	ssize_t ret = read(f, &pcstart, sizeof(uint32));
+	if (ret < 0)
+		fail("failed to read bitmap start");
+	ret = read(f, &pcsize, sizeof(uint32));
```

Comment:
Since we already mmap the whole file, reading few fields separately feels somewhat awkward.
I would do:

```
struct cov_filter_t {
	uint32 pcstart;
	uint32 pcsize;
	uint8 bitmap[];
};

static cov_filter_t* cov_filter;

inline void init_cov_filter()
{
#if SYZ_EXECUTOR_USES_SHMEM
	int f = open("/syz-cover-bitmap", O_RDONLY);
	if (f < 0) {
		// We don't fail here because we don't know yet if we should use coverage filter or not.
		// We will receive the flag only in execute flags and will fail in coverage_filter if necessary.
		debug("bitmap is no found, coverage filter disabled\n");
		return;
	}
	struct stat st;
	if (fstat(f, &st))
		fail("failed to stat coverage filter");
	void* preferred = (void*)0x110f230000ull;
	cov_filter = (cov_filter_t*)mmap(preferred, st.st_size, PROT_READ, MAP_PRIVATE|MAP_ANON, f, 0);
	if (bitmap != preferred)
		fail("failed to initialize bitmap at %p", preferred);
	if (st.st_size < (cov_filter->pcsize + 7) / 8)
		fail(...);
	close(f);
#endif
}
```

It's a bit shorter and simpler and will make any future changes to the format simpler as well (since we have everything in memory already, we can access any fields directly as necessary without additional reads).

Reply:
> if (st.st_size < (cov_filter->pcsize + 7) / 8)

Probably:
if (st.st_size < sizeof(*cov_filter) + (cov_filter->pcsize + 7) / 8))